src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java

```diff
@@ -1,24 +1,620 @@
+        Logger.LOG_EXCEPTION(e);
+        initializeSuccess = false;
+      }
+    }
+
+    // Configure Ewon's HTTP timeouts
+    try {
+      SCHttpUtility.setHttpTimeouts(HTTP_TIMEOUT_SECONDS_STRING);
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Failed to set the Ewon system's HTTP timeout value!");
+      Logger.LOG_EXCEPTION(e);
+      initializeSuccess = false;
+    }
+
+    // Calculate local time offset and configure queue
+    try {
+      SCTimeUtils.injectJvmLocalTime();
+
+      final Date currentTime = new Date();
+      final String currentLocalTime = SCTimeUtils.getIso8601LocalTimeFormat().format(currentTime);
+      final String currentUtcTime = SCTimeUtils.getIso8601UtcTimeFormat().format(currentTime);
+      Logger.LOG_DEBUG(
+          "The local time zone is "
+              + SCTimeUtils.getTimeZoneName()
+              + " with an identifier of "
+              + SCTimeUtils.getLocalTimeZoneDesignator()
+              + ". The current local time is "
+              + currentLocalTime
+              + ", and the current UTC time is "
+              + currentUtcTime
+              + ".");
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Unable to inject local time into the JVM!");
+      Logger.LOG_EXCEPTION(e);
+      initializeSuccess = false;
+    }
+
+    // Configure queue string history data option
+    try {
+      HistoricalDataQueueManager.setStringHistoryEnabled(
+          connectorConfig.getQueueDataStringEnabled());
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL(
+          "Failed to configure the queue option for enabling/disabling string history data!");
+      Logger.LOG_EXCEPTION(e);
+      initializeSuccess = false;
+    }
+
+    // Configure queue data poll size
+    try {
+      HistoricalDataQueueManager.setQueueFifoTimeSpanMins(
+          connectorConfig.getQueueDataPollSizeMinutes());
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL(
+          "Failed to configure the queue data poll size interval (minutes) option!");
+      Logger.LOG_EXCEPTION(e);
+      initializeSuccess = false;
+    }
+
+    // Configure queue diagnostic tags option
+    try {
+      Logger.LOG_CRITICAL(
+          "Configuring the queue diagnostic tags option..."
+              + connectorConfig.getQueueDiagnosticTagsEnabled());
+      CConnectorDataMgr.configureQueueDiagnosticTags(
+          connectorConfig.getQueueDiagnosticTagsEnabled());
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Failed to configure the queue diagnostic tags enabled option!");
+      Logger.LOG_EXCEPTION(e);
+      initializeSuccess = false;
+    }
+
+    if (initializeSuccess) {
+      Logger.LOG_CRITICAL("Finished initializing " + CONNECTOR_FRIENDLY_NAME + ".");
+    } else {
+      Logger.LOG_CRITICAL("Failed to initialize " + CONNECTOR_FRIENDLY_NAME + "!");
+    }
+    return initializeSuccess;
+  }
+
+  /** Method for performing connector application initialization steps. */
```

Comment:
same as the comment for initialize method

Reply:
Not sure what this is in reference to.

Reply:
This is the same comment as the initialize method. line 119.  What is the difference between startup and initialize?

Reply:
The initialize phase is preparation for start up, so things like loading the configuration file, injecting local time, setting HTTP timeout length, etc.

The start up phase is where things that rely on initialization like device provisioning and set up of MQTT occur.

Having these two separate phases removes the need for confusing return statements in the middle of methods, because it allows initialization to fail separately from start up, regular operation, shutdown, etc. Our convention during code reviews has always been to avoid multiple return statements in a method, in favor of a single return.

Reply:
there's nothing confusing about returns in the middle. It only stinks when you set a breakpoint at the last line of the function assuming it's the only return (when you could have set the breakpoint at the caller). If we make this a rule we are doomed to have code with way too many if checks, and executes slower.

Reply:
I'm on the fence about this one. We've followed this rule in most, if not all, of our connector development, so I think at the very least it would be good to decide as a team on a true convention here so we don't create more differences between our code bases.

Reply:
This has come up during previous review(s): 

https://github.com/hms-networks/flexy-azure-connector/pull/57#discussion_r582963250

Reply:
@alexjhawk and @it-hms Are there still breakpoint limitations with the ewon java debugger? I don't recall if they updated that with the switch from ME to SE.  If not there is probably more of an argument to start inserting returns mid method. 

Modern languages with modern tools - use as many mid method/function returns as you would like :)

Reply:
I wasn't aware of this limitation, but I am not sure if it is around still or not. I don't make use of the Ewon debugger very often and usually opt for a significant amount of Logger output instead.

Reply:
I'm all in for use mid function returns when it makes sense. I was going to put it to a vote and bribe everyone that didn't have a strong opinion.

Reply:
Whats the bribe? I can be bought. 

A lot of times unless its like 4 or more nested if statements there is an argument just to break up the functionality into smaller private helper methods.